application/common/models/User.php

```diff
@@ -1026,4 +1030,56 @@ public function extendGracePeriodIfNeeded()
             ]);
         }
     }
+
+    /**
+     * Delete all user records that are inactive and have not been updated recently.
+     */
+    public static function deleteInactiveUsers()
+    {
+        $enabled = \Yii::$app->params['inactiveUserDeletionEnable'];
```

Comment:
I think you forgot to test this value before proceeding, unless it is somewhere else?

Reply:
It's in the query, line 1060 (`andWhere('1=' . (int) $enabled)`)

Reply:
That seems a little too subtle. What would you think of simply dealing with it right at the beginning?
```php
    if ($enabled !== true) {
        \Yii::warning([
            'action' => 'delete inactive users',
            'status' => 'disabled',
        ]);
        return;
    }
```